lib/std/os/windows/user32.zig

```diff
@@ -1,104 +1,100 @@
 usingnamespace @import("bits.zig");
 
-// PM
+pub const WM = enum(u16) {
+    NULL = 0x0000,
+    CREATE = 0x0001,
+    DESTROY = 0x0002,
+    MOVE = 0x0003,
+    SIZE = 0x0005,
+
+    ACTIVATE = 0x0006,
+    PAINT = 0x000F,
+    CLOSE = 0x0010,
+    QUIT = 0x0012,
+    SETFOCUS = 0x0007,
+
+    KILLFOCUS = 0x0008,
+    ENABLE = 0x000A,
+    SETREDRAW = 0x000B,
+
+    SYSCOLORCHANGE = 0x0015,
+    SHOWWINDOW = 0x0018,
+
+    WINDOWPOSCHANGING = 0x0046,
+    WINDOWPOSCHANGED = 0x0047,
+    POWER = 0x0048,
+
+    CONTEXTMENU = 0x007B,
+    STYLECHANGING = 0x007C,
+    STYLECHANGED = 0x007D,
+    DISPLAYCHANGE = 0x007E,
+    GETICON = 0x007F,
+    SETICON = 0x0080,
+
+    INPUT_DEVICE_CHANGE = 0x00fe,
+    INPUT = 0x00FF,
+    KEYDOWN = 0x0100,
+    KEYUP = 0x0101,
+    CHAR = 0x0102,
+    DEADCHAR = 0x0103,
+    SYSKEYDOWN = 0x0104,
+    SYSKEYUP = 0x0105,
+    SYSCHAR = 0x0106,
+    SYSDEADCHAR = 0x0107,
+    KEYLAST = 0x0109,
+
+    COMMAND = 0x0111,
+    SYSCOMMAND = 0x0112,
+    TIMER = 0x0113,
+
+    MOUSEMOVE = 0x0200,
+    LBUTTONDOWN = 0x0201,
+    LBUTTONUP = 0x0202,
+    LBUTTONDBLCLK = 0x0203,
+    RBUTTONDOWN = 0x0204,
+    RBUTTONUP = 0x0205,
+    RBUTTONDBLCLK = 0x0206,
+    MBUTTONDOWN = 0x0207,
+    MBUTTONUP = 0x0208,
+    MBUTTONDBLCLK = 0x0209,
+    MOUSEWHEEL = 0x020A,
+    XBUTTONDOWN = 0x020B,
+    XBUTTONUP = 0x020C,
+    XBUTTONDBLCLK = 0x020D,
+};
+
+pub const WA = enum(u16) {
+    INACTIVE = 0,
+    ACTIVE = 0x0006,
+    ACTIVATE = 0x0006,
+};
+
+pub const SW = enum(u16) {
+    HIDE = 0,
+    SHOW = 5,
+};
+
 pub const PM_REMOVE = 0x0001;
 pub const PM_NOREMOVE = 0x0000;
 pub const PM_NOYIELD = 0x0002;
 
-// WM
-pub const WM_NULL = 0x0000;
-pub const WM_CREATE = 0x0001;
-pub const WM_DESTROY = 0x0002;
-pub const WM_MOVE = 0x0003;
-pub const WM_SIZE = 0x0005;
-
-pub const WM_ACTIVATE = 0x0006;
-pub const WM_PAINT = 0x000F;
-pub const WM_CLOSE = 0x0010;
-pub const WM_QUIT = 0x0012;
-pub const WM_SETFOCUS = 0x0007;
-
-pub const WM_KILLFOCUS = 0x0008;
-pub const WM_ENABLE = 0x000A;
-pub const WM_SETREDRAW = 0x000B;
-
-pub const WM_SYSCOLORCHANGE = 0x0015;
-pub const WM_SHOWWINDOW = 0x0018;
-
-pub const WM_WINDOWPOSCHANGING = 0x0046;
-pub const WM_WINDOWPOSCHANGED = 0x0047;
-pub const WM_POWER = 0x0048;
-
-pub const WM_CONTEXTMENU = 0x007B;
-pub const WM_STYLECHANGING = 0x007C;
-pub const WM_STYLECHANGED = 0x007D;
-pub const WM_DISPLAYCHANGE = 0x007E;
-pub const WM_GETICON = 0x007F;
-pub const WM_SETICON = 0x0080;
-
-pub const WM_INPUT_DEVICE_CHANGE = 0x00fe;
-pub const WM_INPUT = 0x00FF;
-pub const WM_KEYFIRST = 0x0100;
```

Comment:
Keys like WM_KEYFIRST, WM_KEYLAST, WM_MOUSEFIRST, WM_MOUSELAST, etc are needed for functions that take ranges of events, e.g. [GetMessage](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getmessage).  Since they overlap with other values, there are two options:
1) Make this an extern enum, which [allows duplicate values](https://github.com/ziglang/zig/issues/2115)
2) Expose them as constants instead of enum values, i.e. `pub const KEYFIRST = @intToEnum(@This(), 0x0100);` in the enum namespace.

If we're eventually planning to update other structures/function prototypes to use this type instead of `UINT`, it makes sense to make these extern enums anyway.  But since the *FIRST/*LAST aren't actual 'enum values' (you would never make a case for them in a switch statement), it might make sense to do both and keep them as constants rather than enum values.